src/client/options/mod.rs

```diff
@@ -327,6 +390,12 @@ pub struct ClientOptions {
     #[builder(default)]
     pub selection_criteria: Option<SelectionCriteria>,
 
+    /// The declared API version
+    ///
+    /// The default value is to have no declared API version
+    #[builder(default)]
+    pub server_api: Option<ServerApi>,
```

Comment:
In addition to passing this via the options, do we want a specific `with_uri_str` method that also takes a version since it can't be specified via string?

e.g.
```rust
let server_api = ServerApi::builder().version(ServerApiVersion::v1).build();
let client = Client::with_uri_str_and_server_api("mongodb://...", server_api).await?;
```
It's a bit of a mouthful but it's still a little easier than parsing options first.

Reply:
I wasn't sure what kind of factories you'd want to have. I'll defer to @saghm and @isabelatkinson on a decision on this and can implement it if everyone agrees 👍

Reply:
Given that there's already a way to parse a string into options and then pass that, I think I'd prefer not to do that until we have a strong sense that this is something that users want. If we ever have another option that can't be specified in the string, the combinations start to get unwieldy. There's a decent risk that we might wish we had done something more extensible to multiple options later.

Reply:
SGTM

Reply:
> If we ever have another option that can't be specified in the string, the combinations start to get unwieldy.

Just pointing out that FLE will bring `autoEncryptionOpts` which will be the same kind of client option as `serverApi`, so not including option-specific constructors sounds sensible 👍